crates/turborepo-repository/src/change_mapper/mod.rs

```diff
@@ -29,6 +29,21 @@ pub enum LockfileChange {
     ChangedPackages(HashSet<turborepo_lockfiles::Package>),
 }
 
+/// This describes the state of a change to a lockfile.
+pub enum LockfileContents {
+    /// We know the lockfile did not change
+    Unchanged,
+    /// We know the lockfile changed but don't have the file contents of the
+    /// previous lockfile (i.e. `git status`, or perhaps a lockfile that was
+    /// deleted or otherwise inaccessible with the information we have)
```

Comment:
This is a good callout. Fetching the previous lockfile will fail if the user switches package managers as we'll try to read the current package manager's lockfile.

Reply:
thanks!!